lib/Doctrine/ORM/Persisters/UnrecognizedField.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Persisters;
+
+use Doctrine\ORM\PersisterException;
+
+class UnrecognizedField extends \Exception implements PersisterException
+{
+    public static function fromField(string $field): self
```

Comment:
What about a method name like `UnrecognizedField::byName()` or `UnrecognizedField::ofName()`?

Also: missing space in return value at `:`.